### DISCUSSION TITLE
ImportError: cannot import name 'NDArray' from 'numpy.typing' (C:\Users\*user*\Miniconda3\lib\site-packages\numpy\typing\__init__.py)

### DISCUSSION THREAD

**Discussion by DrTrendo** [danielhrisca/asammdf#708] [Q&A] [answered]
Hi all,

I am pretty amaised from the asammdf package and want to try it.
But when I installed it in a virtual environment and tried to import it got the following warnings and errors:

ldf is not supported
xls is not supported
xlsx is not supported
yaml is not supported
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Input In [1], in <cell line: 1>()
----> 1 import asammdf

File ~\Miniconda3\lib\site-packages\asammdf\__init__.py:15, in <module>
     12 logger.setLevel(logging.ERROR)
     14 from .blocks.options import get_global_option, set_global_option
---> 15 from .blocks.source_utils import Source
     16 from .gui import plot
     17 from .mdf import MDF, SUPPORTED_VERSIONS

File ~\Miniconda3\lib\site-packages\asammdf\blocks\source_utils.py:10, in <module>
      6 from __future__ import annotations
      8 from functools import lru_cache
---> 10 from . import v2_v3_blocks as v3b
     11 from . import v2_v3_constants as v3c
     12 from . import v4_blocks as v4b

File ~\Miniconda3\lib\site-packages\asammdf\blocks\v2_v3_blocks.py:26, in <module>
     24 from . import v2_v3_constants as v23c
     25 from ..version import __version__
---> 26 from .utils import get_fields, get_text_v3, MdfException, UINT16_u, UINT16_uf
     28 SEEK_START = v23c.SEEK_START
     29 SEEK_END = v23c.SEEK_END

File ~\Miniconda3\lib\site-packages\asammdf\blocks\utils.py:53, in <module>
     51 import numpy as np
     52 from numpy import arange, bool_, dtype, interp, where
---> 53 from numpy.typing import NDArray
     54 from pandas import Series
     56 from . import v2_v3_constants as v3c

ImportError: cannot import name 'NDArray' from 'numpy.typing' (C:\Users\*user*\Miniconda3\lib\site-packages\numpy\typing\__init__.py)

As I read in a previous discussion the warnings on the top are not critical, but what does this error at the bottom mean?
NumPy is installed in the virtual environment and I am using it without any problems.
I also updated conda and did a clean install in a new virtual environment, which also didn't help.
I am using Python on a Windows machine and do not have admin rights, if this playes any roll. 
Can anyone help me fix this issue?

**Comment by danielhrisca** [comment]
I think you need to update numpy to a newer version. What version do you have?

**Comment by DrTrendo** [comment, marked as answer]
Hi, thanks for the fast answer!
I didn't have bumpy at all, but now I installed it:
 Requirement already satisfied: bumpy in c:\users\*user*\miniconda3\lib\site-packages (0.2.4)
 Note: you may need to restart the kernel to use updated packages.

Even now I get the same error, when importing asammdf

**Reply by danielhrisca** [reply]
:)) sorry for the typo , I meant to write ``numpy``

**Reply by DrTrendo** [reply]
Probably you were right. I removed conda and reinstalled it.
Now I am running numpy 1.22.4 and there is no issue with asammdf.
Thanks!

**Comment by TiHONGTONG** [comment]
你最后是怎么解决的？asmmdf的版本是多少

**Reply by danielhrisca** [reply]
numpy 2.0 is not working yet

**Reply by TiHONGTONG** [reply]
what is  asammdf version and other environment？i try two days，not work
